apollo-federation/src/sources/connect/expand/mod.rs

```diff
@@ -614,6 +614,32 @@ mod helpers {
                         ))
                     }
                 }?;
+            } else {
+                // If we don't have a key, but the type is an interfaceObject
+                // we'll try just copying over the defined keys.
+                let output_ty = output_type.get(self.original_schema.schema())?;
+                let has_interface_object = output_ty
+                    .directives()
+                    .iter()
+                    .any(|d| d.name == name!("federation__interfaceObject"));
+                if has_interface_object {
+                    let keys = output_ty
+                        .directives()
+                        .iter()
+                        .filter(|d| d.name == name!("federation__key"));
+                    for key in keys {
+                        match &output_type {
+                            TypeDefinitionPosition::Object(o) => {
+                                o.insert_directive(to_schema, key.clone())
+                            }
+                            TypeDefinitionPosition::Interface(i) => {
+                                i.insert_directive(to_schema, key.clone())
+                            }
+
+                            _ => Ok(()),
```

Comment:
Do we need to worry about types marked with `@interfaceObject` that aren't actually an interface nor object? I would assume that validation catches that, but maybe it would still be a good idea to return an error here.

Reply:
@interfaceObject's only valid schema location is OBJECT so this cannot happen